askama_derive/src/generator.rs

```diff
@@ -1061,8 +1064,8 @@ impl<'a> Generator<'a> {
             )?;
         }
 
-        // Since nodes must not outlive the Generator, we instantiate
-        // a nested Generator here to handle the include's nodes.
+        // Since nodes must not outlive the Generator, we instantiate a nested `Generator` here to
```

Comment:
Because we pass `nodes` to `child.handle()` and `nodes` need to outlive the child `Generator`, I don't think we can easily share a map of included ASTs between parents and children. This should already be quite the improvement, as at least includes from the same template don't have to reparse.